pandas/_libs/tslib.pyx

```diff
@@ -658,16 +661,15 @@ cdef _array_to_datetime_object(
 
         cnp.PyArray_MultiIter_NEXT(mi)
 
-    if len(unique_timezones) > 1:
-        warnings.warn(
-            "In a future version of pandas, parsing datetimes with mixed time "
-            "zones will raise an error unless `utc=True`. "
-            "Please specify `utc=True` to opt in to the new behaviour "
-            "and silence this warning. To create a `Series` with mixed offsets and "
-            "`object` dtype, please use `apply` and `datetime.datetime.strptime`",
-            FutureWarning,
-            stacklevel=find_stack_level(),
-        )
+    warnings.warn(
+        "In a future version of pandas, parsing datetimes with mixed time "
```

Comment:
So if we call `_array_to_datetime_object` we _always_ know there are mixed timezone components? The comments on line 614-617 doesn't seem to suggest so?

Reply:
correct.  we only get here through the array_to_datetime path with `not is_same_offsets`